compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -118,9 +126,10 @@ public JvmObservabilityGen(JvmPackageGen pkgGen) {
      * @param pkg The package to instrument
      */
     public void rewriteObservableFunctions(BIRPackage pkg) {
+        compileTimeConstants = new HashMap<>();
```

Comment:
Can we move this initialization to constructor? Is this method called only once?

Reply:
Fixed in 759b37d